File: Coding Ninjas - CodeStudio/Basics Of C++ Guided Path/reverse_the_array.cpp
#include <iostream>
using namespace std;

int main() {
	int n;
    cin >> n;
    int arr[n];
    for(int i = 0; i < n; i++)
        cin >> arr[i];
    int x = 0, y = n-1;
    while(x < y) {
        int temp = arr[x];
        arr[x++] = arr[y];
        arr[y--] = temp;
    }
    for(int i = 0; i < n; i++) {
        cout << arr[i] << " ";
    }
    cout << endl;
    return 0;
}

File: Coding Ninjas - CodeStudio/Basics of Java Guided Path/reverse_the_array.java
import java.util.ArrayList;

public class Solution 
{
    public static void reverseArray(ArrayList<Integer> arr, int m)
    {
        int low = m + 1, high = arr.size() - 1;
        while(low < high) {
            int temp = arr.get(low);
            arr.set(low, arr.get(high));
            arr.set(high, temp);
            low++;
            high--;
        }
    }
}
